compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -2557,6 +2557,10 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
                 GenericArg::Const(ct) => {
                     self.visit_anon_const(&ct.value);
                 }
+                GenericArg::Infer(inf) => {
+                    self.visit_id(inf.hir_id);
+                    i += 1;
```

Comment:
now looking at this, I'm not totally sure if just adding +1 for types is correct now that I've changed the type, will dig more into it later

Reply:
I think in this current implementation, because we're only lowering from the AST in the same point as the original implementation, this worked, but I updated it if there are future places to lower.